IS1001Test.py

```diff
@@ -12,12 +12,623 @@
+            data=data,
+            params=parameters,
+            allow_redirects=False,
+            auth=auth,
+            verify=CERT_TRUST_ROOT_CA
+        )
+
+    def _raise_nmos_exception(self, test, response, string=''):
+        """Raise NMOS Exception with HTTP Response Information"""
+        raise NMOSTestException(test.FAIL(
+            """Request to Auth Server failed. {}.
+            URL: {},
+            Request Body: {},
+            Status_code: {},
+            Response: {}"""
+            .format(string, response.url, response.request.body, response.status_code, response.text)
+        ))
+
+    def _verify_response(self, test, expected_status, response):
+        """Verify if HTTP response status code is valid"""
+        if isinstance(response, requests.models.Response):
+            try:
+                response.raise_for_status()
+            except Exception:
+                self._raise_nmos_exception(test, response)
+
+        elif isinstance(response, str):
+            raise NMOSTestException(test.FAIL(
+                "Request to Auth Server failed due to: {}".format(response)
+            ))
+
+        if response.status_code != expected_status:
+            self._raise_nmos_exception(test, response, "Expected status_code: {}".format(expected_status))
+
+    def _verify_redirect(self, test, response, input_params, query_key, expected_query_value=None):
+        """Verify if HTTP response is a valid redirect with given parameters"""
+
+        if not response.is_redirect or response.status_code != 302:
+            raise NMOSTestException(test.FAIL(
+                "Request to server did not result in a redirect. Received {} instead of 302 when requesting '{}'. {}"
+                .format(response.status_code, query_key, response.json())
+            ))
+        if "location" not in response.headers:
+            raise NMOSTestException(test.FAIL(
+                "'Location' not found in Response headers when requesting '{}'. Headers = {}"
+                .format(query_key, response.headers.keys())
+            ))
+        if query_key not in response.headers["location"]:
+            raise NMOSTestException(test.FAIL(
+                "'{}' not found in Location Header. Location Header = {}"
+                .format(query_key, response.headers["location"])
+            ))
+        if "state" not in response.headers["location"]:
+            raise NMOSTestException(test.FAIL(
+                "'state' not found in Location Header. Location Header = {}"
+                .format(response.headers["location"])
+            ))
+
+        redirect_uri, query_string = response.headers["location"].split('?')
+        parsed_query = parse_qs(query_string)
+        actual_query_value = parsed_query[query_key][0]
+        if "state" in parsed_query.keys():
+            state = parse_qs(query_string)["state"][0]
+        else:
+            state = None
+
+        if not redirect_uri == input_params["redirect_uri"]:

```

Comment:
I'd have thought this should just be != rather than not ==, but then I'm surprised the linter didn't catch it so it may not be the case. There are a few cases of this.